pkg/events/events_test.go

```diff
@@ -165,3 +166,32 @@ func TestRecordEventWithCache(t *testing.T) {
 		t.Error("invalid Count")
 	}
 }
+
+func TestEventWithEmptyNamespaceReference(t *testing.T) {
+	clientMock := &k8sClientMock{}
+
+	recorder := events.NewEventRecorder(clientMock, newTestScheme(), events.EventsMap, events.EventRecorderOptions{
+		Version:   "1",
+		Cluster:   "cluster-1",
+		Component: "controller",
+		Namespace: "test",
+	})
+
+	pod := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test-pod",
+		},
+	}
```

Comment:
This is not a good example for test. In Pod spec, namespace is a mandatory field.
Were you able to fix the original issue with Namespace object with this fix? I'm not sure how we can raise events for cluster scoped objects like namespaces or clusterrole.

Reply:
This change fixed the issue while testing locally. Using namespace object instead now.